ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -236,17 +236,16 @@ export class AccountCreation extends React.Component<
     });
 
     if (value === Country.US) {
-      this.updateAddress('country', value);
-
       const stateCodeGuess = this.autoSelectStateCode(
         this.state.profile.address.state
       );
       if (stateCodeGuess != null) {
         this.updateAddress('state', stateCodeGuess);
       }
-    } else {
-      this.updateAddress('country', '');
     }
+
+    // OTHER country values are set by the 'Other country input' TextInput
+    this.updateAddress('country', value === Country.OTHER ? '' : value);
```

Comment:
This is a way to express the intentions of the old logic too, which was implemented at a time when US and OTHER were the only options